lib/deimos/active_record_consume/batch_record_list.rb

```diff
@@ -23,6 +23,24 @@ def filter!(method)
         self.batch_records.delete_if { |record| !method.call(record.record) }
       end
 
+      # @param method [Proc]
+      # @param block [Block]
+      # @return [Array<BatchRecord>]
+      def reject!(method=nil, &block)
+        if method.nil?
+          self.batch_records.reject!(&block)
+        else
+          case method.parameters.size
+          when 2
+            self.batch_records.select! do |record|
+              method.call(record.record, record.associations)
+            end
+          else
+            self.batch_records.select! { |record| method.call(record.record) }
+          end
+        end
+      end
```

Comment:
Not sure I understand why we need this complex method? Also it's super confusing - in regular Ruby, `delete_if` and `reject!` are almost identical. And `reject!` is the *oppposite* of `select!`, so... what's going on?

Reply:
This should all be `reject`` 😞 No clue what brain farts led to this

Reply:
Changed this to use `partition` instead